File: vuejs/exercise2/numbers.test.ts
import { expect, test } from 'vitest'
import { NumbersInWords } from "./numbers";

test.each([
    [1, "one"],
    [2, "two"],
    [9, "nine"],
    [19, "nineteen"],
    [20, "twenty"],
    [21, "twenty one"],
    [99, "ninety nine"],
    [100, "one hundred"],
    [135, "one hundred and thirty five"]
])("%i prints as %s", (number, expected) => {
    expect(NumbersInWords.printNumber(number)).toBe(expected);
})

File: .samples/dotnet/baby_steps/NumbersTest.cs
using Xunit;

// Kata instructions: https://www.codewars.com/kata/54fa4e210609868fce0002bf

namespace Numbers 
{
    public class NumbersTest
    {
        [Theory]
        [InlineData(1, "one")]
        [InlineData(2, "two")]
        [InlineData(19, "nineteen")]
        [InlineData(20, "twenty")]
        [InlineData(21, "twenty one")]
        [InlineData(99, "ninety nine")]
        [InlineData(100, "one hundred")]
        [InlineData(999, "nine hundred and ninety nine")]
        public void PrintNumberUniversalTest(int input, string expected)
        {
            Assert.Equal(expected,new NumbersInWords().PrintNumber(input));
        }
    }

}